utils/providers.py

```diff
@@ -1,27 +1,26 @@
 """ Helper functions related to the creation, listing, filtering and destruction of providers
 
-The functions in this module that require the 'filters' parameter, such as list_providers,
-setup_a_provider etc depend on a (by default global) dict of filters by default.
-If you are writing tests or fixtures, you want to depend on those functions as de facto gateways.
+The list_providers function in this module depend on a (by default global) dict of filters.
+If you are writing tests or fixtures, you want to depend on this function as a de facto gateway.
 
-The rest of the functions, such as get_mgmt, get_crud etc ignore this global dict and will provide
-you with whatever you ask for with no limitations.
+The rest of the functions, such as get_mgmt, get_crud, get_provider_keys etc ignore this global
+dict and will provide you with whatever you ask for with no limitations.
 
 The main clue to know what is limited by the filters and what isn't is the 'filters' parameter.
 """
 import operator
-import random
 import six
 from collections import Mapping, OrderedDict
 from copy import copy
 
-from fixtures.pytest_store import store
+
 from cfme.common.provider import BaseProvider
 
-from cfme.containers import provider as containers_providers # NOQA
-from cfme.cloud import provider as cloud_providers # NOQA
-from cfme.infrastructure import provider as infrastructure_providers # NOQA
-from cfme.middleware import provider as middleware_providers # NOQA
+
+from cfme.containers import provider as containers_providers  # NOQA
```

Comment:
This won't be required soon because of entrypoints ;) Oh what a glorious day that will be ;)